internal/app/machined/internal/phase/config/extra_files.go

```diff
@@ -31,12 +31,34 @@ func (task *ExtraFiles) TaskFunc(mode runtime.Mode) phase.TaskFunc {
 	return task.runtime
 }
 
+// nolint: gocyclo
 func (task *ExtraFiles) runtime(r runtime.Runtime) (err error) {
-	var result *multierror.Error
+	var (
+		content string
+		result  *multierror.Error
+	)
 
 	for _, f := range r.Config().Machine().Files() {
-		// Slurp existing file if append is our op and add contents to it
-		if f.Op == "append" {
+		switch f.Op {
+		case "create":
+			if err = existsAndIsFile(f.Path); err == nil {
+				return fmt.Errorf("file must not exist: %q", f.Path)
+			}
+		case "overwrite":
+			if err = existsAndIsFile(f.Path); err != nil {
+				return err
+			}
+
+			if err = os.Remove(f.Path); err != nil {
```

Comment:
can we remove from readonly rootfs? or we can just bind mount on top of that?

Reply:
We cannot. Was hoping that when we attempt to remove that the error would bubble up and it would be clear to the user. Should we verify the path?

Reply:
Oh I see. Your saying don't remove, and just let the logic after the `switch` to its' thing.

Reply:
yep, as it will bind mount overwriting it effectively I guess in any case